HeterogeneousCore/MPICore/plugins/api.h

```diff
@@ -3,6 +3,8 @@
 
 // externals headers
 #include <mpi.h>
+#include <iostream>
+#include <typeinfo>
```

Comment:
I think you can remove `<iostream>`.
If we do need `<typeinfo>`, could you reformat as
```suggestion
// C++ standard library headers
#include <typeinfo>

// externals headers
#include <mpi.h>
```
?